src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.LastLineCache.cs

```diff
@@ -17,59 +18,54 @@ internal partial class TagComputer
             /// <summary>
             /// it is a helper class that encapsulates logic on holding onto last classification result
             /// </summary>
-            private class LastLineCache
+            private class LastLineCache : ForegroundThreadAffinitizedObject
             {
                 // this helper class is primarily to improve active typing perf. don't bother to cache
                 // something very big. 
                 private const int MaxClassificationNumber = 32;
-                private readonly object _gate = new();
```

Comment:
this type was only used from GetTags.  That is only on the UI thread, so that invariant falls over to this type.  this allows us to simplify by not needing a gate.